internal/app/fipcontroller/hcloud.go

```diff
@@ -54,27 +54,29 @@ func (controller *Controller) servers(ctx context.Context, ips []net.IP) (server
 }
 
 // Search for a hetzner Server that has the given ip in any of its networks
-func (controller *Controller) searchServerForIP(servers []*hcloud.Server, ip net.IP) *hcloud.Server {
-	for _, server := range servers {
-		// IP must not be a floating IP, but might be private or public depending on the cluster configuration
-		if server.PublicNet.IPv4.IP.Equal(ip) || server.PublicNet.IPv6.IP.Equal(ip) {
-			controller.Logger.Debugf("Found matching public IP on server '%s'", server.Name)
-			return server
-		}
+func (controller *Controller) searchServerForIP(servers []*hcloud.Server, ips []net.IP) *hcloud.Server {
+	for _, ip := range ips {
+		for _, server := range servers {
+			// IP must not be a floating IP, but might be private or public depending on the cluster configuration
+			if server.PublicNet.IPv4.IP.Equal(ip) || server.PublicNet.IPv6.IP.Equal(ip) {
+				controller.Logger.Debugf("Found matching public IP on server '%s'", server.Name)
+				return server
+			}
 
-		hasPrivateNet := false
-		for _, privateNet := range server.PrivateNet {
-			if privateNet.IP.Equal(ip) {
-				hasPrivateNet = true
-				break
+			hasPrivateNet := false
+			for _, privateNet := range server.PrivateNet {
+				if privateNet.IP.Equal(ip) {
+					hasPrivateNet = true
```

Comment:
`note` Technically you can return the server without assigning and checking `hasPrivateNet` first